packages/mds-policy-service/tests/policy-client.spec.ts

```diff
@@ -0,0 +1,306 @@
+    })
+
+    it('will not edit or delete a published Policy', async () => {
+      await PolicyServiceClient.writePolicy(ACTIVE_POLICY_JSON)
+      await PolicyServiceClient.publishPolicy(ACTIVE_POLICY_JSON.policy_id, ACTIVE_POLICY_JSON.publish_date)
+      const publishedPolicy = clone(ACTIVE_POLICY_JSON)
+      publishedPolicy.name = 'a shiny new name'
+      await expect(PolicyServiceClient.editPolicy(publishedPolicy)).rejects.toThrow()
+      await expect(PolicyServiceClient.deletePolicy(publishedPolicy.policy_id)).rejects.toThrow()
+    })
+
+    it('will throw an error if attempting to edit a nonexistent Policy', async () => {
+      const policy = clone(POLICY2_JSON)
+      policy.policy_id = '28218022-d333-41be-bda5-1dc4288516d2'
+      await expect(PolicyServiceClient.editPolicy(policy)).rejects.toThrowError(NotFoundError)
+    })
+  })
+
+  describe('unit test PolicyMetadata functions', () => {
+    beforeAll(async () => {
+      await PolicyRepository.initialize()
+    })
+
+    beforeEach(async () => {
+      await PolicyRepository.deleteAll()
+    })
+
+    afterAll(async () => {
+      await PolicyRepository.shutdown()
+    })
+
+    it('.readBulkPolicyMetadata', async () => {
+      await PolicyServiceClient.writePolicy(ACTIVE_POLICY_JSON)
+      await PolicyServiceClient.writePolicy(POLICY2_JSON)
+      await PolicyServiceClient.writePolicy(POLICY3_JSON)
+
+      await PolicyServiceClient.writePolicyMetadata({
+        policy_id: ACTIVE_POLICY_JSON.policy_id,
+        policy_metadata: { name: 'policy_json' }
+      })
+      await PolicyServiceClient.writePolicyMetadata({
+        policy_id: POLICY2_JSON.policy_id,
+        policy_metadata: { name: 'policy2_json' }
+      })
+      await PolicyServiceClient.writePolicyMetadata({
+        policy_id: POLICY3_JSON.policy_id,
+        policy_metadata: { name: 'policy3_json' }
+      })
+
+      const noParamsResult = await PolicyServiceClient.readBulkPolicyMetadata({})
+      expect(noParamsResult.length).toStrictEqual(3)
+      const withStartDateResult = (await PolicyServiceClient.readBulkPolicyMetadata({
+        start_date: now(),
+        get_published: null,
+        get_unpublished: null
+      })) as PolicyMetadataDomainModel<{ name: string }>[]
+      expect(withStartDateResult.length).toStrictEqual(1)
+      expect(withStartDateResult[0].policy_metadata?.name).toStrictEqual('policy3_json')
+
+      const meta = await PolicyServiceClient.readSinglePolicyMetadata(ACTIVE_POLICY_JSON.policy_id)
+      expect(meta.policy_id).toStrictEqual(ACTIVE_POLICY_JSON.policy_id)
+    })
+
+    it('updates policy metadata', async () => {
+      await PolicyServiceClient.writePolicy(ACTIVE_POLICY_JSON)
+      const metadata = await PolicyServiceClient.writePolicyMetadata({
+        policy_id: ACTIVE_POLICY_JSON.policy_id,
+        policy_metadata: { name: 'policy_json' }
+      })
+      await PolicyServiceClient.updatePolicyMetadata({
```

Comment:
This should read back the metadata to make sure it was updated.

Reply:
and there was a bug too! thanks, fixed.